include/llbuild/Basic/Clock.h

```diff
@@ -0,0 +1,47 @@
+//===- Clock.h --------------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_CORE_CLOCK_h
+#define LLBUILD_CORE_CLOCK_h
+
+#include <algorithm>
+#include <chrono>
+
+namespace llbuild {
+namespace basic {
+
+class Clock {
+public:
+  /// A timestamp is the number of seconds since the clock's epoch time.
+  typedef double Timestamp;
+  
+  Clock() LLBUILD_DELETED_FUNCTION;
+  
+  /// Returns a global timestamp that represents the current time in seconds since a reference data.
+  /// *NOTE*: This function uses a monotonic clock, so don't compare between systems.
+  inline static Timestamp now() {
+    // steady_clock is monotonic
+    auto now = std::chrono::steady_clock::now();
+    std::chrono::duration<double> difference = std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch());
+    return difference.count();
+  }
+  
+  /// Returns the relative duration in seconds since the given timestamp.
+  inline static double durationSince(Timestamp other) {
+    return now() - other;
+  }
```

Comment:
I think we can probably skip this function, and just inline the math (which is actually shorted, and self explanatory).

Reply:
It‘s actually not used so far but I wanted to use this for the rule results API in my other PR. But I guess you‘re right that it‘s self explanatory.